fw/http.c

```diff
@@ -2572,6 +2572,33 @@ tfw_http_set_hdr_date(TfwHttpMsg *hm)
 	return r;
 }
 
+/*
+ * Add 'Upgrade:' header for websocket upgrade messages
+ */
+static int
+tfw_http_set_hdr_upgrade(TfwHttpMsg *hm, bool is_resp)
+{
+	int r = 0;
+
+	if (test_bit(TFW_HTTP_B_UPGRADE_WEBSOCKET, hm->flags)) {
+		if (is_resp && ((TfwHttpResp *)hm)->status == 101
+		    && test_bit(TFW_HTTP_B_UPGRADE_EXTRA, hm->flags))
```

Comment:
I'm not sure if I understand this. It seems this is about https://datatracker.ietf.org/doc/html/rfc7230#section-6.7

> A server that sends a 101 (Switching Protocols) response MUST send an
   Upgrade header field to indicate the new protocol(s) to which the
   connection is being switched; if multiple protocol layers are being
   switched, the sender MUST list the protocols in layer-ascending
   order.  A server MUST NOT switch to a protocol that was not indicated
   by the client in the corresponding request's Upgrade header field.
 
This piece of code is for response with something extra plus to `websocket` in `Upgrader` header. Since we can only generate `Upgrade: websocket` in adjusted client request and server MUST NOT add anything non-requested, then what could be in the response for the extra? Can't we just always send `Upgrade: websocket` from the server to the client?

If I'm wrong, then a good comment with the RFC cite is required in the place.

Reply:
My intention was as follows:

We always strip upgrade header from response and re-add it when needed. So we do not just pass it through tempesta as is. But when we recreate headers would be beneficially to do simple sane check because we essentially hide backend and act on behalf on it.

Imaging erroneous backend that we on our discretion silently transform into compliant backend. That can would be destructive if you ask me.

But this intention is, i understand, subtle and subjective. It is up to you if we need just alway send `Upgrade: websocket`.